lib/preface_make/bounded_meet_semilattice.mli

```diff
@@ -0,0 +1,35 @@
+(** Building a {!module:Preface_specs.Bounded_meet_semilattice} *)
+
+(** {1 Using the minimal definition}
+
+    Build a {!module-type:Preface_specs.BOUNDED_MEET_SEMILATTICE} using
+    {!module-type:Preface_specs.Bounded_meet_semilattice.WITH_MEET}. Standard
+    method, using the minimal definition of a semigroup to derive its full API. *)
+
+module Via_meet (Req : Preface_specs.Bounded_meet_semilattice.WITH_MEET) :
```

Comment:
I think that you can some modules  (that take a semilattice and a module with top): 

- `Core_over_meet_semilattice`
- `Over_meet_semilattice`

Reply:
Yes, you can take some inspiration of the contravariant hierarchy. It is a little bit verbose but it is mainly to try, at least, to keep the API consistent.
OK for you @hakimba?

Reply:
Ok !

Reply:
Hum, i'm not sure that everything is good. I suggest : 

```ocaml
module type WITH_TOP = sig
  type t
  val top : t
end

module type WITH_MEET_AND_TOP = sig
  include WITH_TOP
  include Meet_semilattice.WITH_MEET with type t := t
end

module type CORE = WITH_MEET_AND_TOP
```
```ocaml
module Core_via_meet_and_top
    ( Preface_specs.Bounded_meet_semilattice.WITH_MEET_AND_TOP)
    : Preface_specs.Bounded_meet_semilattice.CORE
     
module Core_over_meet_semilattice_and_via_top
   (Preface_specs.Meet_semilattice.CORE)
   (Preface_specs.Bounded_meet_semilattice.WITH_TOP)
   : Preface_specs.Bounded_meet_semilattice.CORE
   
(* Usual functors for Infix, Operation and Syntax (if needed). *)

module Via_meet_and_top 
    ( Preface_specs.Bounded_meet_semilattice.WITH_MEET_AND_TOP)
    : Preface_specs.Bounded_meet_semilattice.API
    
module Over_meet_semilattice_and_via_top
   (Preface_specs.Meet_semilattice.CORE)
   (Preface_specs.Bounded_meet_semilattice.WITH_TOP)
   : Preface_specs.Bounded_meet_semilattice.API
```

Something like that.